ui/dashboards/src/components/EditDashboardButton/EditDashboardButton.tsx

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Button } from '@mui/material';
+import PencilIcon from 'mdi-material-ui/PencilOutline';
+
+export interface EditDashboardButtonProps {
+  /**
+   * The label used inside the button.
+   */
+  label?: string;
+
+  /**
+   * Handler that puts the dashboard into editing mode.
+   */
+  onClick: () => void;
+}
+
+export const EditDashboardButton = ({ label = 'Edit', onClick }: EditDashboardButtonProps) => {
```

Comment:
Some thinking:

There's nothing in this component that's actually specific to editing a dashboard (it's a `Button` where the fixed properties are variant, color, and startIcon -- in other words, it's an outlined, grey button that always has a `PencilIcon`)

Given that, I think calling it `EditButton` might make sense